src/test/llmq_signing_shares_tests.cpp

```diff
@@ -0,0 +1,342 @@
+        }
+
+        // Create members
+        std::vector<CDeterministicMNCPtr> members;
+        for (int i = 0; i < size; ++i) {
+            // For simplicity, use nullptr members since we're testing pre-verification logic
+            // The actual member checking happens in IsMember() which we can't fully mock
+            members.push_back(nullptr);
+        }
+
+        quorum->Init(std::move(qc_ptr), nullptr, InsecureRand256(), members);
+
+        // Set verification vector if requested
+        if (hasVerificationVector) {
+            std::vector<CBLSPublicKey> vvec;
+            for (int i = 0; i < size; ++i) {
+                CBLSSecretKey sk;
+                sk.MakeNewKey();
+                vvec.push_back(sk.GetPublicKey());
+            }
+            quorum->SetVerificationVector(vvec);
+        }
+
+        return quorum;
+    }
+
+    // Helper to create test SessionInfo
+    CSigSharesNodeState::SessionInfo CreateTestSessionInfo(CQuorumCPtr quorum)
+    {
+        CSigSharesNodeState::SessionInfo session;
+        session.llmqType = quorum->params.type;
+        session.quorumHash = quorum->qc->quorumHash;
+        session.id = InsecureRand256();
+        session.msgHash = InsecureRand256();
+        session.quorum = quorum;
+        return session;
+    }
+
+    // Helper to create test BatchedSigShares
+    CBatchedSigShares CreateTestBatchedSigShares(const std::vector<uint16_t>& members)
+    {
+        CBatchedSigShares batched;
+        batched.sessionId = 1;
+
+        for (uint16_t member : members) {
+            CBLSLazySignature lazySig;
+            batched.sigShares.emplace_back(member, lazySig);
+        }
+
+        return batched;
+    }
+};
+
+BOOST_FIXTURE_TEST_SUITE(llmq_signing_shares_tests, LLMQSigningSharesTestFixture)
+
+// Test: Missing verification vector
+// Note: This test will likely return QuorumTooOld or NotAMember because we can't fully mock
+// IsQuorumActive and IsMember. However, we can still verify the function is callable and
+// returns a proper PreVerifyBatchedResult.
+BOOST_AUTO_TEST_CASE(preverify_missing_verification_vector)
+{
+    // Create quorum WITHOUT verification vector
+    auto quorum = CreateMinimalTestQuorum(3, false);
+    auto sessionInfo = CreateTestSessionInfo(quorum);
+    auto batchedSigShares = CreateTestBatchedSigShares({0, 1});
+
+    // Call PreVerifyBatchedSigShares - it should detect missing verification vector
+    // (if it gets past the earlier checks for quorum activity and membership)
+    auto result = CSigSharesManager::PreVerifyBatchedSigShares(*mn_activeman, *Assert(m_node.llmq_ctx->qman),
+                                                               sessionInfo, batchedSigShares);
+
+    // We expect it to fail (not be successful)
+    BOOST_CHECK(!result.IsSuccess());
+    // The actual result will likely be QuorumTooOld, NotAMember, or MissingVerificationVector
+    // depending on which check fails first
```

Comment:
Shouldn't it be deterministic? Why having a test like that if we can't get the expected `MissingVerificationVector` error?